src/api/app/controllers/webui/search_controller.rb

```diff
@@ -163,5 +163,9 @@ def search
                                 issue_tracker_name: @search_tracker)
     @results = search.search(page: params[:page], per_page: @per_page)
     flash[:notice] = 'Your search did not return any results.' if @results.empty?
+  rescue ThinkingSphinx::ConnectionError => e
+    flash[:error] = 'Error connecting to the search service.'
```

Comment:
Not very helpful for you customers. What can they do about it? :)

And what about all the other exceptions that could occur?

Reply:
> I think our customers can do nothing except wait until we fix whatever is the reason for the connection error. It's on our side to detect sphinx downtimes via monitoring and bring it back.

That's what you should tell them then.